src/fhirtypes/ElementDefinition.ts

```diff
@@ -1819,13 +1823,30 @@ export class ElementDefinition {
     ) {
       let newElements: ElementDefinition[] = [];
       if (this.contentReference) {
-        // TODO: Implement recursive constraints once the approach is better clarified.
-        // See: https://chat.fhir.org/#narrow/stream/179252-IG-creation/topic/Clarification.20on.20contentReference
-
         // Get the original resource JSON so we unfold unconstrained reference
         const type = this.structDef.type;
         const json = fisher.fishForFHIR(type, Type.Resource);
-        if (json) {
+        // contentReference elements will not contain a type field, so we must fish for the StructDef and
+        // check the differential
+        const profileJson = fisher.fishForFHIR(this.structDef.id, Type.Profile);
+        if (this.hasProfileElementExtension(profileJson)) {
+          const def = StructureDefinition.fromJSON(profileJson);
```

Comment:
It isn't clear to me that this is necessary. From what I am seeing `this.structDef` does already have the element with the required constraints on it. I'm testing with this example:
```
Profile: MyQ
Parent: Questionnaire
* item.code 0..2
* item ^type[0].profile[0] = http://example.org/StructureDefinition/MyQ
* item ^type[0].profile[0].extension[0].url = http://hl7.org/fhir/StructureDefinition/elementdefinition-profile-element
* item ^type[0].profile[0].extension[0].valueString = "Questionnaire.item"
* item.item.code 0..3
```
And I'm finding that just doing:
```
          const def = this.structDef;
```
works the same for me, and the new tests pass as well. Let me know if you have an example of why we really do need to fish for the profile.

Reply:
`this.structdef` does in fact have the element with the required constraints on it, however that element does not contain the profile-element extension on it's `type.profile` property as the `_profile` array is not present (shown in the attached screenshot). To check and see if the extension is there, I found that I had to fish for that StructureDefinition separately and then check the differential array.  

There's no need to convert that JSON back into a `StructureDefinition` on line 1833 as I'm currently doing, but I do think there's a need to fish for the `profileJson` in order to check for the extension and know to unfold from the element on the current profile, rather than the parent profile.
<img width="897" alt="Screen Shot 2021-05-11 at 3 56 39 PM" src="https://user-images.githubusercontent.com/53443041/117876585-88bbf500-b271-11eb-9f85-b45497a9d921.png">